test/distributed/checkpoint/test_state_dict.py

```diff
@@ -3,12 +3,15 @@
 import copy
 import sys
 from itertools import chain
-from typing import Callable, Tuple
+from typing import Callable, Tuple, Type, Union
 
 import torch
 import torch.distributed as dist
 import torch.nn as nn
 from torch.distributed._composable import fully_shard, replicate
+# importing fully_shard as FSDP2 since the original fully_shard is used in this test.
+# TODO: remove old composable fully_shard so that we don't have to import new fully_shard as FSDP2
```

Comment:
Sorry about this! We should add some deprecation warnings to old composable `fully_shard` and try to remove it from internal.

Reply:
> Sorry about this! We should add some deprecation warnings to old composable `fully_shard` and try to remove it from internal.

No worry! I just don't want to remove the test, while the old APIs are still around. I can add the deprecation warning in the original composable API.